Flickr/Sources/FlickrAppVCs/FavoritesCoordinator.swift

```diff
@@ -34,33 +40,45 @@ class FavoritesCoordinator {
 
 extension FavoritesCoordinator: FavoritesViewControllerDelegate {
     func selectedImageFromFavoritesVC(imageData: Data, imageTitle: String, imageId: String) {
-        let photoViewerVC = PhotoViewerVC(url: nil, imageTitle: imageTitle, imageId: imageId, imageData: imageData)
+        let photoViewerVC = PhotoViewerVC(url: nil, imageTitle: imageTitle, imageId: imageId, imageData: imageData, dbPool: dbPool)
         photoViewerVC.title = ""
         photoViewerVC.favoritesDelegate = self
         rootNavigationController.pushViewController(photoViewerVC, animated: true)
     }
 
     func getFavoriteImagesFromStorage() -> [FavoriteImageData]? {
-        PersistenceManager.retrieveData()
+        var favoriteimagesArray: [FavoriteImageData] = []
+        do {
+            try dbPool.read { db in
+                let array = try Row.fetchCursor(db, sql: "SELECT * FROM favorites")
+                while let item = try array.next() {
+                    favoriteimagesArray.append(FavoriteImageData(imageId: item["id"], imageData: item["image"], imageTitle: item["name"]))
+                }
+            }
+            return favoriteimagesArray
+        } catch {
+            return nil
+        }
     }
 
     func storeImageAsFavorite(imageData: Data, id: String, title: String) {
-        var favorites = FavoriteImagesArray(array: [])
-        favorites.array = PersistenceManager.retrieveData() ?? []
-        let image = FavoriteImageData(imageId: id, imageData: imageData, imageTitle: title)
-        favorites.array.append(image)
-        PersistenceManager.storeData(favorites)
+        do {
+            try dbPool.write { db in
+                try db.execute(sql: "INSERT INTO favorites (id, name, image) VALUES (?, ?, ?)", arguments: [id, title, imageData])
```

Comment:
Please use FetchableRecords and PersistableRecords for writing the image data to database. This ensures Typesafetly while writing and reading the data from the database